### DISCUSSION TITLE
Cant connect to Channel

### DISCUSSION THREAD

**Discussion by ZytechTV** [discord-net/Discord.Net#2698] [Q&A] [answered]
I'm currently facing an issue with my Discord bot where it's unable to retrieve the desired channel.  
`var channel = _client.GetChannel(channelId) as SocketTextChannel;`
 Channel is null after. 
I have taken the necessary steps to troubleshoot the problem. I verified that the channel ID I provided is accurate and checked the bot's permissions to ensure it has the required access. Additionally, I made sure to add the bot to the Discord server and confirmed that it has the necessary permissions to access the server and the channel.
 
**Programm Class**
```
namespace Zygakure
{
    public class Program
    {
        public static void Main(string[] args)
        {
            DiscordBot bot = new DiscordBot();
            bot.MainAsync().GetAwaiter().GetResult();
        }
    }
}
```

**DiscordBot Class**
```
using Discord.WebSocket;
using Discord;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Zygakure
{
    public class DiscordBot
    {
        public async Task MainAsync()
        {
            DiscordSocketClient _client = new DiscordSocketClient();

            _client.Log += Log;

            // Discord Token
            string token = "MTExNjU5ODY2MDA5NzExNDIyMg.Gi9mCl.K2DmsXggvkxXKnFFv5DlXcoKwok4woQGB8YxNk";

            await _client.LoginAsync(TokenType.Bot, token);
            await _client.StartAsync();

            _client.MessageReceived += HandleMessageReceived;

            // Channel
            ulong channelId = 1116600689100083240;
            var channel = _client.GetChannel(channelId) as SocketTextChannel;

            if (channel == null)
            {
                Console.WriteLine("Error at getting Channel");
                return;
            }
            
            await Task.Delay(-1);
        }

        private Task Log(LogMessage message)
        {
            Console.WriteLine(message.ToString());
            return Task.CompletedTask;
        }

        private async Task HandleMessageReceived(SocketMessage message)
        {            
            Console.WriteLine($"Input: {message.Content}");
                        
            if (message.Content.Contains("Test"))
            {
                await message.Channel.SendMessageAsync("Test done");
            }
        }
    }
}
```

Someone has an idea ? :) Greetings

PS: Token and Channel IDs are fake in this Code

**Comment by gehongyan** [comment, marked as answer]
@tokle002 
> https://discordnet.dev/guides/getting_started/first-bot.html#creating-a-discord-client

> We may now invoke the client's [StartAsync](https://discordnet.dev/api/Discord.WebSocket.DiscordSocketClient.html#Discord_WebSocket_DiscordSocketClient_StartAsync_) method, which will start connection/reconnection logic. It is important to note that **this method will return as soon as connection logic has been started!** Any methods that rely on the client's state should go in an event handler. This means that you should not directly be interacting with the client before it is fully ready.

Try getting the channel inside the handlers subscribed to [Ready](https://discordnet.dev/api/Discord.WebSocket.DiscordSocketClient.html#Discord_WebSocket_DiscordSocketClient_Ready) event.

```csharp
_client.Ready += () =>
{
    var channel = _client.GetChannel(1116600689100083240) as SocketTextChannel;
    return Task.CompletedTask;
};
```

**Reply by Anheledir** [reply]
And you should create a new bot token, never post these anywhere publicly!

**Reply by ZytechTV** [reply]
It is connecting now and responding to my inputs, but it is not displaying the inputs.

**Changed Code**
```
 _client.Ready += () =>
            {
                var channel = _client.GetChannel(channelId) as SocketTextChannel;
                _client.Log += Log;

                if (channel == null)                
                    Console.WriteLine("Error at getting Channel");         
                
                return Task.CompletedTask;
            };
```

**Output**
```
09:02:47 Discord     Discord.Net v3.10.0 (API v10)
09:02:47 Gateway     Connecting
09:02:48 Gateway     You're using the GuildScheduledEvents gateway intent without listening to any events related to that intent, consider removing the intent from your config.
09:02:48 Gateway     You're using the GuildInvites gateway intent without listening to any events related to that intent, consider removing the intent from your config.
09:02:48 Gateway     Connected
09:02:49 Gateway     Ready
09:02:49 Gateway     Ready
Input:
```

**Reply by Anheledir** [reply]
You'll need to enable the message content intent in your code and in the discord developer portal. Otherwise the bot can only see the content of messages it is mentioned in and in DMs.

**Reply by IOKG04** [reply]
How do I enable it?

**Comment by mwpowellhtx** [comment]
How do we get a list of said channels back, their names, and Ids included. For UI configuration purposes, for instance. Thanks!
